synapse/storage/databases/main/registration.py

```diff
@@ -1357,12 +1357,15 @@ def _use_registration_token_txn(txn):
             # Override type because the return type is only optional if
             # allow_none is True, and we don't want mypy throwing errors
             # about None not being indexable.
-            res: Dict[str, Any] = self.db_pool.simple_select_one_txn(
-                txn,
-                "registration_tokens",
-                keyvalues={"token": token},
-                retcols=["pending", "completed"],
-            )  # type: ignore
+            res = cast(
+                Dict[str, Any],
+                self.db_pool.simple_select_one_txn(
+                    txn,
+                    "registration_tokens",
+                    keyvalues={"token": token},
+                    retcols=["pending", "completed"],
+                ),
+            )
```

Comment:
I had to remind myself why we needed a cast here -- it is because `simple_select_one_txn` is a `staticmethod` and overrides don't work for it.

It seems the other methods that do this are class methods instead....